timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -704,5 +710,30 @@ private DbBulkRequest extractFailedRequestsFromBulk(DbBulkRequest dbBulkRequest,
 		return dbBulkRequest;
 	}
 
+	public boolean isWithPersistence() {
+		return withPersistence;
+	}
+
+	public boolean retryFailedRequestsFromDisk() {
+		boolean keepRunning = false;
+		if (withPersistence && diskHandler.hasFailedBulks()){
+			keepRunning = true;
+			LOG.info("------------------ Failed Requests From Disk Retry Start ------------------");
+			List<DbBulkRequest> failedRequestsFromDisk = diskHandler.fetchAndDeleteFailedBulks();
+			for (DbBulkRequest dbBulkRequest : failedRequestsFromDisk) {
+				if (!sendDbBulkRequest(dbBulkRequest, 0)){
+					keepRunning = false;
+				}
```

Comment:
if a job fails to retry a request then it won't keep fetching bulks from disk so there will be a break until the next job will be executed.
Two points:
1.  in the code above we take a break after one failed bulks. do u think we should choose a higher limit?
2. in some cases, this break can be very short (a short time after the break, 10 minutes passed from the last job and a new one is executed). we can sleep for some time so the next job will wait

Reply:
1. I think one fail is good enough
2. No need for a sleep